### PR TITLE
ddl: allow enum/set to append values

### LINKED ISSUE #6398: Support modify column to enum types.
TiDB :
```
CREATE TABLE ti1(a INT NOT NULL, b INT, c INT) engine=InnoDB;
ALTER TABLE ti1 ADD COLUMN e ENUM('a', 'b') FIRST;
ALTER TABLE ti1 MODIFY COLUMN e ENUM('a', 'b', 'c');
```

reports `Error 1105: unsupported modify column modify enum column is not supported`

### PR DESCRIPTION
**Pull request by lqs**
## What have you changed? (mandatory)
Allow to append new values to existing ENUM/SET fields.
Removing values or changing existing values is still unsupported.

## What are the type of the changes (mandatory)?
- Improvement (non-breaking change which is an improvement to an existing feature)
Fix #6398

## How has this PR been tested (mandatory)?
Added unit test cases in ddl/ddl_db_test.go

### CODE REVIEW

**Comment by shenli** [comment]
@lqs Thanks for your PR!

**Comment by coocood** [comment]
@lqs 
Thank you for your contribution!

Would you please add a test for the `Set` type as well?

**Comment by shenli** [comment]
@zimulala PTAL

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -1356,7 +1356,21 @@ func modifiable(origin *types.FieldType, to *types.FieldType) error {
...
+			for index, originElem := range origin.Elems {
+				toElem := to.Elems[index]
+				if originElem != toElem {
+					msg := fmt.Sprintf("cannot modify value %s to %s", originElem, toElem)
+					return errUnsupportedModifyColumn.GenByArgs(msg)
+				}
+			}
+			return nil
+		}
 		return errUnsupportedModifyColumn.GenByArgs("modify enum column is not supported")
```
Comment (winoros) [inline ddl/ddl_api.go]: I think this error msg is not clear enough for the new code.

File: ddl/ddl_db_test.go [hunk 2]
```diff
@@ -1244,6 +1244,7 @@ func (s *testDBSuite) TestChangeColumn(c *C) {
 	c.Assert(hasNotNull, IsFalse)
 	// for enum
 	s.mustExec(c, "alter table t3 add column en enum('a', 'b', 'c') not null default 'a'")
+	s.mustExec(c, "alter table t3 modify column en enum('a', 'b', 'c', 'd') not null default 'a'")
```
Comment (zimulala) [inline ddl/ddl_db_test.go]: Could you add some tests about insert or update statements after this test？

**Comment by mail2fish** [comment]
Is anyone still working on this issue?

**Comment by coocood** [comment]
@mail2fish 
Would you like to help update this PR?

**Comment by mail2fish** [comment]
Yes, I am working on it.

**Comment by coocood** [comment]
@mail2fish 
Alright

**Comment by winkyao** [comment]
Duplicated with https://github.com/pingcap/tidb/pull/7767.

**Pull request closed without merging**